### PR TITLE
doc comments for Storage.*

### PR DESCRIPTION
**Pull request by rowanmiller**
This is all provider-facing API, so I tried to focus on concisely describing the purpose of each class.

### CODE REVIEW

File: src/EntityFramework.Core/Update/IUpdateEntry.cs [hunk 1]
```diff
@@ -8,19 +8,88 @@
...
+        /// </summary>
+        /// <param name="propertyBase"> The property to get or set the current value for. </param>
+        /// <returns> The current value assigned to the property. </returns>
         object this[[NotNull] IPropertyBase propertyBase] { get; [param: CanBeNull] set; }
 
+        /// <summary>
+        ///     Gets an object representing the values currently assigned to the primary key.
+        ///     <see cref="IKeyValue"/> is typically used to test the equivalence of key values. 
+        /// </summary>
+        /// <param name="originalValue"> To be removed. </param>
```
Comment (rowanmiller) [inline src/EntityFramework.Core/Update/IUpdateEntry.cs]: I'll be sending a PR to remove `originalValue` parameters shortly.

Comment (rowanmiller) [inline src/EntityFramework.Core/Update/IUpdateEntry.cs]: @ajcvickers turns out we do actually use these to get key values for deleted entities https://github.com/aspnet/EntityFramework/blob/dev/src/EntityFramework.Relational/Update/Internal/CommandBatchPreparer.cs#L216.

File: src/EntityFramework.Core/Storage/DatabaseErrorLogState.cs [hunk 2]
```diff
@@ -4,11 +4,19 @@
...
 
 namespace Microsoft.Data.Entity.Storage
 {
+    /// <summary>
+    ///     Provided as the state to <see cref="ILogger"/> when logging an exception that occurred while accessing the database.
+    /// </summary>
     public class DatabaseErrorLogState
     {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="Database" /> class.
```
Comment (lajones) [inline src/EntityFramework.Core/Storage/DatabaseErrorLogState.cs]: Should reference cref="DatabaseErrorLogState" instead.

File: src/EntityFramework.Core/Storage/ValueBuffer.cs [hunk 3]
```diff
@@ -7,18 +7,41 @@
 
 namespace Microsoft.Data.Entity.Storage
 {
+    /// <summary>
+    ///     <para>  
+    ///         Represents a set of indexed values. Typically used to represent a row of data returned from the database.
```
Comment (anpete) [inline src/EntityFramework.Core/Storage/ValueBuffer.cs]: "a database"

File: src/EntityFramework.Core/Storage/IDatabaseCreator.cs [hunk 4]
```diff
@@ -6,11 +6,56 @@
...
+    ///     <para>  
+    ///         This interface is typically used by database providers (and other extensions). It is generally  
+    ///         not used in application code.  
+    ///     </para>  
+    /// </summary>
     public interface IDatabaseCreator
     {
+        /// <summary>
+        ///     Ensures that the database for the context does not exist. If it does not exist, no action is taken. If it does
+        ///     exist then the database is deleted. The entire database is deleted an not effort is made to remove just the
```
Comment (ajcvickers) [inline src/EntityFramework.Core/Storage/IDatabaseCreator.cs]: Typo: "not" => "no". Also, maybe consider prefixing with the word "Warning:"?

**Comment by ajcvickers** [comment]
:shipit:

**Pull request merged**